unix/tx/TXWindow.cxx

```diff
@@ -132,7 +132,7 @@ TXGlobalEventHandler* TXWindow::setGlobalEventHandler(TXGlobalEventHandler* h)
 
 void TXWindow::getColours(Display* dpy, XColor* cols, int nCols)
 {
-  bool* got = new bool[nCols];
+  bool got[nCols];
```

Comment:
this is not valid C++

Reply:
Can you share what's wrong with it?

Reply:
VLAs (arrays with dynamic size allocated on stack) are not part of C++ standard and were even made optional in the recent C standard. They can also lead to issues if the size passed is not sanitized/can be big (you can easily overflow stack).

I'm not sure what specific requirements this project has (was just lurking), but if you don't want to deal with with manual new/delete (it looks atrocious here and already creates plenty of problem), the best solution is to just use `std::vector<...>`.

Reply:
Crap, forgot C++ doesn't have that. I agree that a `std::vector` is probably appropriate here.